### PR TITLE
Add schema for deployment integrations and centralize integrations schemas.

### PR DESCRIPTION
**Pull request by Ferroin**
##### Summary

- Relocate collector schemas under `integrations/schemas`: This is being done as we will be adding more integration schemas that don’t logically belong under `collectors` but having them all in one place makes more sense than splitting them to different places.
- Move truly shared definitions into their own schema file, with proper formatting as a schema, so that they can be used in multiple schemas.
- Add a schema for deployment integrations, based on the requirements document.
- Add a basic `deploy.yaml` file with a few sample deployments.

##### Test Plan

n/a

##### Additional Information

@ilyam8 is flagged for review on this because he’s been actively involved in handling of the collector schemas.

### CODE REVIEW

File: integrations/deploy.yaml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+        - channel: nightly
+          command: >
+            curl https://my-netdata.io/kickstart.sh > /tmp/netdata-kickstart.sh && sh /tmp/netdata-kickstart.sh
+            --nightly-channel --claim-token {% claim_token %} --claim-url {% claim_url %}
+        - channel: stable
+          command: >
+            curl https://my-netdata.io/kickstart.sh > /tmp/netdata-kickstart.sh && sh /tmp/netdata-kickstart.sh
+            --stable-channel --claim-token {% claim_token %} --claim-url {% claim_url %}
+  additional_info: &ref_containers >
+    Did you know you can also deploy Netdata on your OS using Kubernetes or Docker?
```
Comment (sashwathn) [inline integrations/deploy.yaml:34]: We need to link Kubernetes and Docker sections to the relevant Deploy sections

**Review by sashwathn — APPROVED** [review]

File: integrations/deploy.yaml [hunk 2]
```diff
@@ -0,0 +1,48 @@
+- &linux
+  id: deploy-linux-generic
+  meta: &linux_meta
+    name: Linux
+    link: ''
+    categories:
+      - deploy.operating-systems
```
[the remark this answers is not in the document]
Comment (Ferroin) [reply integrations/deploy.yaml:7]: Yes, it’s already there.

File: integrations/deploy.yaml (hunk 1 above)
Comment (Ferroin) [reply integrations/deploy.yaml:34]: I’m not certain what format is going to be needed for that, so I just copied the text for now.

We’re going to need to ask @kapantzak what’s needed for this.

**Review by sashwathn — APPROVED** [review]

File: integrations/deploy.yaml [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    icon_filename: ''
+  keywords:
+    - linux
+  description: 'Run the following command on your node to install and claim Netdata:"r'
+  methods:
+    - method: wget
```
Comment (thiagoftsm) [inline integrations/deploy.yaml:13]: I cannot see `wget2` in the list, should we also start recommend it?

Comment (sashwathn) [reply integrations/deploy.yaml:13]: @thiagoftsm: We don't yet use wget2 for kickstart install and has never been positioned on the Cloud. Do we have this supported and is it backwards compatible with wget?

Comment (thiagoftsm) [reply integrations/deploy.yaml:13]: Hello @sashwathn ,

Yes, it its compatible and the command line to save our kickstart locally is more or less the same:

```sh
$ wget2 -O /tmp/netdata-kickstart.sh https://my-netdata.io/kickstart.sh
```

I believe I already talked about it with our @netdata/agent-sre when it was released.

This is not a blocker, it was more a check.

Best regards!

**Review by thiagoftsm — APPROVED** [review]
I am approving considering that necessary changes to add another software are out of scope of this PR.

**Comment by sashwathn** [comment]
@Ferroin : I had a discussion with @kapantzak and @hugovalente-pm and we want to render the existing `Add Nodes` modal using the same `integrations.js` source.
Can we add another property to the `deploy schema` which is one of `Quick-start` or `Generic-deployment` to mark *Linux, Windows, macOS, Docker, Kubernetes* etc to show up as the only options here.

Also, it was pointed out that the `priority` field is kind of necessary to control the ordering of this modal as well and we will discuss this on the call later today.

image

File: integrations/deploy.yaml (hunk 1 above)
Comment (kapantzak) [reply integrations/deploy.yaml:34]: @Ferroin we should use a custom tag there.

I think both of the links (`Kubernetes`, `Docker`) are going to do the same thing (redirect user to the `Docker & Kubernetes` category) right?

So just use this:

```
Did you know you can also deploy Netdata on your OS using {% GoToCategory categoryId="deploy.docker-kubernetes" %}Kubernetes{% /GoToCategory %} or {% GoToCategory categoryId="deploy.docker-kubernetes" %}Docker{% /GoToCategory %}?
```

**Comment by kapantzak** [comment]
> @Ferroin : I had a discussion with @kapantzak and @hugovalente-pm and we want to render the existing `Add Nodes` modal using the same `integrations.js` source. Can we add another property to the `deploy schema` which is one of `Quick-start` or `Generic-deployment` to mark _Linux, Windows, macOS, Docker, Kubernetes_ etc to show up as the only options here.
> 
> Also, it was pointed out that the `priority` field is kind of necessary to control the ordering of this modal as well and we will discuss this on the call later today.
> 
> Things we would like to complete today:
> 
> * Finalised schema with the `Quick-start` and `Priority` fields.
> * Updated `deploy.yaml` so @kapantzak can start working on the rendering.
> 
> cc: @hugovalente-pm @shyamvalsan
> 
> image

Just mentioning that the prop name should ideally not use parenthesis (`-`), instead we can use underscore:
```
quick_start
```

File: integrations/deploy.yaml (hunk 1 above)
Comment (kapantzak) [reply integrations/deploy.yaml:34]: @Ferroin a small change I mage to the above, better use lower case tag name:

```
GoToCategory --> goToCategory
```

File: integrations/deploy.yaml (hunk 3 above)
Comment (Ferroin) [reply integrations/deploy.yaml:13]: I strongly suspect that users who are conciously choosing to use `wget2` will be smart enough to handle this.

That said, we _may_ want to add support to the kickstart script itself for this, though that should a completely separate PR.

**Pull request merged**